composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/DispatcherProvider.kt

```diff
@@ -0,0 +1,19 @@
+package sp.bvantur.tasky.core
```

Comment:
Putting this in core kinda breaks the layering you have here, since the way you put it doesn't make it part of a specific layer. And if it's not part of a specific layer, what is the rule behind using this `DispatcherProvider`? With presentation/domain/data we have clear rules what can be used where, but here you escape that.

Every single function/class you have can be attributed to one specific layer (presentation, domain or data). Even if you have util functions or classes that works. Few examples:
```
Date formatting util -> core/presentation/util
Success/error wrapper -> core/domain/util
HttpClientExt -> core/data/util
EncryptedSharedPreferencesUtil -> core/data/util
```
Therefore, strictly speaking the `DispatcherProvider` interface would be part of domain and the implementation of data (since that's where implementation details go)

Reply:
So there is no way to have a component that can be used in multiple layers? For example, if you want to write tests for ViewModel you would need to ensure having the same dispatcher provider also in the Presentation layer. And also definitely in the data layer where the dispatcher switch usually happens for IO communication. What to do in such scenario?

Reply:
If your DispatcherProvider is in core->domain, it can be used across the project, since every layer is allowed to access domain, so I'm not sure if I'm getting your question correctly.

Reply:
It's ok....I don't know why I asked that. It makes sense that a dispatcher provider can be used across the project.